### ISSUE TITLE
xml2cpp - generate async property getters

### ISSUE THREAD

**Issue by br337** [Kistler-Group/sdbus-cpp#331]
When generating proxies using xml2cpp, all methods described as async using the async annotation in the XML description of the interface will be converted to async functions in cpp and have their associated callbacks registered. 

In the case of the auto-generated property getters, these use the `getProperty(propertyName).onInterface(INTERFACE_NAME)` pattern. By looking under the hood, one can see that the `onInterface(INTERFACE_NAME)` method builds on top of a *synchronous* function call and thus doesn't work asynchronously, nor does it register a callback. The getter uses the `callMethod("Get")` function on the `org.freedesktop.DBus.Properties` interface under the hood. This, in the case one would want a completely asynchronous program, would become a problem that needs to be fixed, preferably on top of the auto-generated header files by using inheritance and overrides. This would in turn just obfuscate the code even more and add unneeded complexity. 

An option would be the addition of one more key value pair to every Property node in the XML definition of the interface. An example would look like this:
```xml
<property name="Drivers" type="as" access="read" async="true"/>
```

This extra key-value pair would then in turn tell xml2cpp that it needs to get this property using `callMethodAsync()` and register a callback for the getter function.

**Comment by sangelovic** [comment]
Adding this to my TODO list

**Comment by febra33** [comment]
https://github.com/febra33/sdbus-cpp/commit/2ec59cc52ac58327ec34395fe4f203c1d37a92a4

I quickly hacked this function into my fork since I needed it. It works without any problems.

**Comment by febra33** [comment]
I added the needed feature in a fork. 
You can find it here https://github.com/Kistler-Group/sdbus-cpp/commit/f4df7252b83d493df5b82d05f98c9767f5391332
If you think this is looking fine I can create a PR.

**Comment by sangelovic** [comment]
Hey @br337. I took a wider, holistic approach, and implemented support for async property get/set functionality on both convenience and IDL-generated API level. I also made some things more consistent. Also, I extended documentation on properties in sdbus-c++ tutorial. I's there: https://github.com/Kistler-Group/sdbus-cpp/pull/354. Feel free to review and let me know if those changes work fine for you.
